enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,956 @@
+    Name string `json:"name"`
+
+    // action specifies actions to perform on headers, such as setting or deleting headers.
+    // +kubebuilder:validation:Required
+    Action IngressControllerHTTPHeaderActionUnion `json:"action"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaderActionUnion specifies an action to take on an HTTP header.
+// +union
+type IngressControllerHTTPHeaderActionUnion struct {
+    // type defines the type of the action to be applied on the header.
+    // Possible values are Set and Delete.
+    // Set allows you to set http request and response headers.
+    // Delete allows you to delete http request and response headers
+    // +unionDiscriminator
+    // +kubebuilder:validation:Enum:=Set;Delete
+    // +kubebuilder:validation:Required
+    Type IngressControllerHTTPHeaderActionType `json:"type"`
+
+    // set specifies how the HTTP header should be set.
+    // If this field is empty, no action is taken.
+    // +optional
+    // +unionMember
+    Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaderActionType defines actions that can be performed on HTTP headers.
+type IngressControllerHTTPHeaderActionType string
+
+const (
+    // Set specifies that an HTTP header should be set.
+    Set IngressControllerHTTPHeaderActionType = "Set"
+    // Delete specifies that an HTTP header should be deleted.
+    Delete IngressControllerHTTPHeaderActionType = "Delete"
+)
+```
+
+The `IngressControllerSetHTTPHeader` type has a field for specifying a header value.
+```go
+// IngressControllerSetHTTPHeader defines the value which needs to be set on an HTTP header.
+type IngressControllerSetHTTPHeader struct {
+    // value specifies a header value.
+    // Dynamic values can be added. The value will be interpreted as an HAProxy format string as defined in
+    // http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and
+    // otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2
+    // Sample fetchers allowed are "req.hdr" and "ssl_c_der".
+    // Converters allowed are "lower" and "base64".
+    // Examples of header value - "%[req.hdr(host),lower]", "%[req.hdr(host),base64]", "{+Q}[ssl_c_der,base64]", "{+Q}[ssl_c_der,lower]"
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=4096
+    Value string `json:"value"`
+}
+```
+
+B) Setting or deleting HTTP request or response headers via `Route.spec`.
+```go
+type Route struct {
+...
+    // spec is the desired state of the route
+    // +kubebuilder:validation:XValidation:rule="!has(self.tls) || self.tls.termination != 'passthrough' || !has(self.httpHeaders)",message="header actions are not permitted when tls termination is passthrough."
```

Comment:
https://github.com/openshift/api/pull/1364/files#r1255016441

Reply:
your suggestion won't work as mentioned there

Reply:
That will fail.